reactive/calico.py

```diff
@@ -145,41 +228,52 @@ def install_calico_service():
         'ip': get_bind_address(),
         'calico_node_image': hookenv.config('calico-node-image')
     })
-    set_state('calico.service.installed')
-
-
-@when('calico.service.installed')
-@when_not('calico.service.started')
-def start_calico_service():
-    ''' Start the calico systemd service. '''
-    status_set('maintenance', 'Starting calico-node service.')
-    service_start('calico-node')
+    check_call(['systemctl', 'daemon-reload'])
+    service_restart('calico-node')
     service('enable', 'calico-node')
-    set_state('calico.service.started')
+    set_state('calico.service.installed')
 
 
 @when('calico.binaries.installed', 'etcd.available',
-      'calico.etcd-credentials.installed')
+      'calico.etcd-credentials.installed',
+      'leadership.set.calico-v3-data-ready')
 @when_not('calico.pool.configured')
 def configure_calico_pool():
     ''' Configure Calico IP pool. '''
     status_set('maintenance', 'Configuring Calico IP pool')
-    etcd = endpoint_from_flag('etcd.available')
-    env = os.environ.copy()
-    env['ETCD_ENDPOINTS'] = etcd.get_connection_string()
-    env['ETCD_KEY_FILE'] = ETCD_KEY_PATH
-    env['ETCD_CERT_FILE'] = ETCD_CERT_PATH
-    env['ETCD_CA_CERT_FILE'] = ETCD_CA_PATH
+
+    # remove unrecognized pools
```

Comment:
Could the user have some pools they created that shouldn't be deleted?

Reply:
We don't support user-created pools, currently, but we will soon with the BGP work.

My original design for Calico BGP called for an `enable-default-pool` config. It made sense to me to change that to `manage-pools` and have it cover the deletion of unrecognized pools, too. That's what I'm going for.

It is possible to have this code only delete pools with conflicting CIDRs. The logic for that isn't too bad actually. But I figure with the `enable-default-pools` / `manage-pools` config coming up soon, it's not really needed.

Reply:
As long as we have it worked out how a user can create and manage pools, I'm ok with this. I just wanted to ensure we thought about the upgrade scenario.